batch/batch/batch_format_version.py

```diff
@@ -0,0 +1,50 @@
+from hailtop.batch_client.aioclient import Job
+
+
+class BatchFormatVersion:
+    def __init__(self, format_version):
+        self.format_version = format_version
+
+    def has_full_spec_in_gcs(self):
+        return self.format_version > 1
+
+    def has_full_status_in_gcs(self):
+        return self.format_version > 1
+
+    def has_full_status_in_db(self):
+        return self.format_version == 1
+
+    def has_attempt_in_log_path(self):
+        return self.format_version > 1
+
+    def db_spec(self, spec):
+        if self.format_version == 1:
+            return spec
+
+        return {
+            'secrets': spec.get('secrets'),
+            'service_account': spec.get('service_account'),
+            'has_input_files': len(spec.get('input_files', [])) > 0,
+            'has_output_files': len(spec.get('output_files', [])) > 0,
+            'attributes': spec.get('attributes')
+        }
```

Comment:
I think we can do the same thing here: store a 5-array instead of a dict.

We might also consider not duplicating the attributes in the job spec and query them from the jobs attributes table.  This would mean attributes would be like the full status, we wouldn't include them in query jobs, but we would include them in the job detail.  I'd be OK with that.  That would further shrink the spec and make this a 4-array.  Finally, we should store the has_... fields as integers (0 or 1) to make them smaller.

Reply:
Are booleans not the same as integers?

Reply:
We could also reduce it further by making the secrets and service account data be tuples/lists.

Reply:
Nope!  JSON has boolean types:

```
>>> json.dumps({'foo': True})
'{"foo": true}'
>>> json.dumps({'foo': 0})
'{"foo": 0}'
```

 > We could also reduce it further by making the secrets and service account data be tuples/lists.

Yes, I considered suggesting that.  In general, these should only be used by CI so I wasn't too worried about them.  For all user pipelines we should make sure they are null.

Reply:
But if you want to convert them I'm all for it!